src/management/JobsManager.js

```diff
@@ -135,42 +135,46 @@ JobsManager.prototype.importUsers = function(data, cb) {
   var url = options.baseUrl + '/jobs/users-imports';
   var method = 'POST';
 
-  var promise = new Promise(function(resolve, reject) {
-    request(
-      {
-        url: url,
-        method: method,
-        headers: headers,
-        formData: {
-          users: {
-            value: data.users_json ? Buffer.from(data.users_json) : fs.createReadStream(data.users),
-            options: {
-              filename: data.users_json ? 'users.json' : data.users,
-            }
-          },
-          connection_id: data.connection_id
+  var promise = options.tokenProvider.getAccessToken().then(function(access_token) {
```

Comment:
Before the existing request promise, we call `getAccessToken` from the provider we were passed on creation. This is then passed a function that wraps the existing code.

Reply:
I *think* this is the right way to do it, but I'm not an expert with the various ways this codebase handles authentication, so if it should do something different as well / instead, please let me know.